defaults for saved query name and description

jrchudy opened the pull request:
This PR adds default values into the name and description columns during saved query creation. These values will populate the form when it is opened and can be replaced by other values by the user.

This also fixes an issue with markdown inputs not displaying correctly in the saved query popup that hasn't been encountered yet.

RFSH reviewed (changes requested):
If our goal here is to use what users see in the chiclets, then you shouldn't use the raw stable facet. The stable facet is going to give you the "raw values", so it won't apply any formatting and in case of entity modes it will be the raw value instead of rowname.

You should do similar things that we did for chiclets. You can use `vm.reference.facetColumns[i].displayname` to get the displayname of a facet, and `vm.facetModels[i].appliedFilters` to get the list of filters that are applied to that facet. You can see how this is done [here](https://github.com/informatics-isi-edu/chaise/blob/master/common/templates/recordset.html#L94,L118)

RFSH commented on common/table.js:
```diff
@@ -1527,10 +1527,85 @@
+                    });
+
+                    if (scope.vm.search) {
+                        name += " " + scope.vm.search;
```
I think you need a separator between search and the rest.

jrchudy replied:
fixed

RFSH commented on common/table.js:
```diff
@@ -1527,10 +1527,85 @@
+                            } else {
+                                str += " " + name;
+                            }
+                            if (idx+1 != options.length) str += ", "
```
This `", "` is causing two whitespaces after comma, is that intentional?

jrchudy commented on common/table.js:
```diff
@@ -1527,10 +1527,85 @@
+                    var facetTextLengthThreshold = 60; // default to 50 characters;
+                    var nameLengthThreshold = 200; // default to 200 characters
+
+                    var nameDescriptionPrefix = scope.vm.reference.table.name + " with";
```
fixed

jrchudy commented:
I added chaise config properties for the thresholds, added an explanation in the code for the default name/description, and added chaise config documentation. We can decide on the chaise config property names with hongsuda in chaise meeting.

jrchudy commented:
Closing this pull request since #2134 also includes changes for saved query functionality and is based off of this branch.
